policy/policy_handler/legacy/image_packages_by_digest.go

```diff
@@ -0,0 +1,156 @@
+
+	// language=graphql
+	vulnerabilitiesByPackageQuery = `
+	query ($context: Context!, $packageUrls: [String!]!) {
+		vulnerabilitiesByPackage(context: $context, packageUrls: $packageUrls) {
+			purl
+			vulnerabilities {
+			cvss {
+				severity
+				score
+			}
+			fixedBy
+			publishedAt
+			source
+			sourceId
+			updatedAt
+			url
+			vulnerableRange
+			}
+		}
+	}`
+)
+
+type (
+	Package struct {
+		Licenses  []string `edn:"licenses,omitempty"` // only needed for the license policy evaluation
+		Name      string   `edn:"name"`
+		Namespace string   `edn:"namespace"`
+		Version   string   `edn:"version"`
+		Purl      string   `edn:"purl"`
+		Type      string   `edn:"type"`
+	}
+
+	ImagePackagesByDigestResponse struct {
+		ImagePackagesByDigest *ImagePackagesByDigest `json:"imagePackagesByDigest" edn:"imagePackagesByDigest"`
+	}
+
+	ImagePackagesByDigest struct {
+		ImagePackages ImagePackages `json:"imagePackages" edn:"imagePackages"`
+	}
+
+	ImagePackages struct {
+		Packages []Packages `json:"packages" edn:"packages"`
+	}
+
+	Packages struct {
+		Package PackageWithLicenses `json:"package" edn:"package"`
+	}
+
+	PackageWithLicenses struct {
+		Licenses        []string        `json:"licenses" edn:"licenses"`
+		Name            string          `json:"name" edn:"name"`
+		Namespace       *string         `json:"namespace" edn:"namespace"`
+		Version         string          `json:"version" edn:"version"`
+		Purl            string          `json:"purl" edn:"purl"`
+		Type            string          `json:"type" edn:"type"`
+		Vulnerabilities []Vulnerability `json:"vulnerabilities" edn:"vulnerabilities"`
+	}
+
+	Vulnerability struct {
+		Cvss            Cvss    `json:"cvss"`
+		FixedBy         *string `json:"fixedBy"`
+		PublishedAt     string  `json:"publishedAt"`
+		Source          string  `json:"source"`
+		SourceID        string  `json:"sourceId"`
+		UpdatedAt       string  `json:"updatedAt"`
+		URL             *string `json:"url"`
+		VulnerableRange string  `json:"vulnerableRange"`
+	}
+
+	Cvss struct {
+		Severity *string  `json:"severity"`
+		Score    *float32 `json:"score"`
+	}
+
+	VulnerabilitiesByPackageResponse struct {
+		VulnerabilitiesByPackage []VulnerabilitiesByPackage `json:"vulnerabilitiesByPackage"`
+	}
+
+	VulnerabilitiesByPackage struct {
+		Purl            string          `json:"purl"`
+		Vulnerabilities []Vulnerability `json:"vulnerabilities"`
+	}
+)
+
+func MockImagePackagesByDigest(ctx context.Context, req skill.RequestContext, sbomPkgs []Package) (ImagePackagesByDigestResponse, error) {
+	// separated for testing
+	ds, err := data.NewSyncGraphqlDataSource(ctx, req)
+	if err != nil {
+		return ImagePackagesByDigestResponse{}, err
+	}
+
+	return mockImagePackagesByDigest(ctx, req, sbomPkgs, ds)
+}
+
+func mockImagePackagesByDigest(ctx context.Context, req skill.RequestContext, sbomPkgs []Package, ds data.DataSource) (ImagePackagesByDigestResponse, error) {
```

Comment:
Do we definitely need all that? Could we not just convert the `Package` objects into a fake `imagePackagesByDigest` response?

Reply:
OK, it's messy, but should do the job for now. Just need to make sure we test the local eval heavily when we release the new vuln skills.